docs/plans/2025-11-19-content-extraction-pipeline-design.md

```
@@ -0,0 +1,602 @@
+
+@dataclass
+class ProcessorResult:
+    output_path: Path        # Path to processed file on disk
+    filename: str            # Updated filename (e.g., "design.gds" from "design.gds.gz")
+    size_bytes: int          # Final file size
+    metadata: dict[str, Any] # Processing metadata for debugging
+```
+
+### Fixed Three-Stage Pipeline
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)